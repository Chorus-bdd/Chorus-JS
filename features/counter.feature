Uses: StepRegistry
Uses: Selenium

Feature: I can use the counter

  #! StepRegistry start
  Feature-Start:
    Given I open Chrome

  Background:
    Given I navigate to http://localhost:9000
    And StepRegistry client SimpleStepPublisher is connected

  Scenario: I can read the counter
    Then the counter value is 0

  Scenario: I can increment the counter
    Given the counter value is 0
    When I click on the increment button
    Then the counter value is 1

  Scenario: I can decrement the counter
    Given the counter value is 0
    When I click on the decrement button
    Then the counter value is -1
