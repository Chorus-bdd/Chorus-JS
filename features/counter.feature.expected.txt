Feature: I can use the counter                                                                       
  Scenario: Feature-Start
    Given I start a web socket server                                                                            PASSED  
    And I open Chrome                                                                                            PASSED  
    And I navigate to http://localhost:9999                                                                      PASSED  
    And I wait for the web socket client SimpleStepPublisher                                                     PASSED  
  Scenario: I can read the counter
    Given I reset the counter                                                                                    PASSED  
    Then the counter value is 0                                                                                  PASSED  0
  Scenario: I can increment the counter
    Given I reset the counter                                                                                    PASSED  
    And the counter value is 0                                                                                   PASSED  0
    When I click on the increment button                                                                         PASSED  
    Then the counter value is 1                                                                                  PASSED  1
  Scenario: I can decrement the counter
    Given I reset the counter                                                                                    PASSED  
    And the counter value is 0                                                                                   PASSED  0
    When I click on the decrement button                                                                         PASSED  
    Then the counter value is -1                                                                                 PASSED  -1


Features  (total:1) (passed:1) (failed:0)
Scenarios (total:3) (passed:3) (failed:0)
Steps     (total:14) (passed:14) (failed:0) (undefined:0) (pending:0) (skipped:0)
